### PR TITLE
fix federated token config in helm chart

### PR DESCRIPTION
**Pull request by nojnhuh**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it: This PR fixes the federated token config in the Helm chart added in #4809 to resolve the error:
```
Deployment.apps "cloud-controller-manager" is invalid: spec.template.spec.containers[0].volumeMounts[3].name: Not found: "projectedToken"
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fixed helm chart install error due to volume name mismatch for 'projectedToken'
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Comment by marosset** [comment]
/cherry-pick release-1.11
We are seeing similar failures in the sig-windows jobs which are mostly using release-1.11
/cc @jsturtevant

**Comment by CecileRobertMichon** [comment]
@marosset this is a cloud-provider-azure PR, not CAPZ

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4226 is the workaround CAPZ PR

This one should not require cherry-picking

File: helm/cloud-provider-azure/Chart.yaml [hunk 1]
```diff
@@ -13,4 +13,4 @@ maintainers:
 name: cloud-provider-azure
 sources:
 - https://github.com/kubernetes-sigs/cloud-provider-azure
-version: 1.28.1
+version: 1.28.2
```
Comment (nojnhuh) [inline helm/cloud-provider-azure/Chart.yaml:16]: @jackfrancis Should this not bump the chart version if it doesn't change the kubernetes version?

Comment (CecileRobertMichon) [reply helm/cloud-provider-azure/Chart.yaml:16]: https://github.com/kubernetes-sigs/cloud-provider-azure/tree/master/helm/cloud-provider-azure#how-to-pick-an-appropriate-new-version-when-updating-the-helm-chart

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**